jormungandr-watchdog/examples/stdin_echo.rs

```diff
@@ -83,20 +90,109 @@ impl Service for StdoutWriter {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum LogLevel {
+    INFO,
+    WARN,
+    DEBUG,
+    ERROR,
+    TRACE,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct LoggerConfig {
+    level: LogLevel,
+}
+
+impl Default for LoggerConfig {
+    fn default() -> Self {
+        LoggerConfig {
+            level: LogLevel::ERROR,
+        }
+    }
+}
+
+fn log_level_into_tracing_level(level: &LogLevel) -> tracing::Level {
+    return match level {
+        LogLevel::INFO => tracing::Level::INFO,
+        LogLevel::WARN => tracing::Level::WARN,
+        LogLevel::DEBUG => tracing::Level::DEBUG,
+        LogLevel::ERROR => tracing::Level::ERROR,
+        LogLevel::TRACE => tracing::Level::TRACE,
+    };
+}
+
+impl Settings for LoggerConfig {
+    fn add_cli_args<'a, 'b>() -> Vec<Arg<'a, 'b>> {
+        vec![Arg::with_name("Log level")
+            .short("ll")
+            .long("log_level")
+            .takes_value(true)
+            .default_value("Warning")
+            .value_name("LOG_LEVEL")
+            .help("Services log level: []")]
+    }
+
+    fn matches_cli_args<'a>(&mut self, matches: &ArgMatches<'a>) {
+        if let Some(level) = matches.value_of("cfg") {
+            self.level = match level.to_lowercase().as_str() {
+                "info" => LogLevel::INFO,
+                "warn" => LogLevel::WARN,
+                "debug" => LogLevel::DEBUG,
+                "error" => LogLevel::ERROR,
+                "trace" => LogLevel::TRACE,
+                _ => self.level.clone(),
```

Comment:
This line should not return a default level. If you use clap's `possible_values` you can expect that if this parameter is used then it cannot have invalid values in it.

```suggestion
                _ => unreachable!(),
```